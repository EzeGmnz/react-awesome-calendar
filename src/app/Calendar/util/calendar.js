const firstDayOfWeek = 0;
const lastDayOfWeek = 6;

const returnDifferenceInDaysBetweenTwoDate = (start, end) => {
    const timeDiff = Math.abs(end.getTime() - start.getTime());
    return Math.ceil(timeDiff / (1000 * 3600 * 24));
};

const getStartOfCalendar = (date) => {
    let originalStartDate = new Date(date.getTime());
    let actualStartDate = new Date(date.getTime());

    if (originalStartDate !== firstDayOfWeek) {
        actualStartDate.setDate(originalStartDate.getDate() - originalStartDate.getDay() + firstDayOfWeek);
    }
    return actualStartDate;
};
const getEndOfCalendar = (date) => {
    let originalEndDate = new Date(date.getTime());
    let actualEndDate = new Date(date.getTime());

    if (originalEndDate !== 7 - 1 - firstDayOfWeek) {
        actualEndDate.setDate(originalEndDate.getDate() - originalEndDate.getDay() + lastDayOfWeek);
    }
    return actualEndDate;

};

export const getCalendar = (month, year) => {
    let originalStartDate = new Date(year, month, 1);
    let originalEndDate = new Date(year, month + 1, 0);
    let actualStartDate = getStartOfCalendar(originalStartDate);
    let actualEndDate = getEndOfCalendar(originalEndDate);

    let dayDifference = returnDifferenceInDaysBetweenTwoDate(actualStartDate, actualEndDate);

    let calendar = [];

    for (let x = 0; x < dayDifference / 7; x++) {
        let week = [];
        for (let y = 0; y < 7; y++) {
            let currentDate = new Date(actualStartDate.getTime());
            currentDate.setDate(actualStartDate.getDate() + (x * 7) + y);
            week.push({
                current: currentDate.getMonth() === originalStartDate.getMonth(),
                date: currentDate
            });
        }
        calendar.push(week);
    }
    return calendar;
};

export const getMonthName = (month) => {
    const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
    return months[month];
};