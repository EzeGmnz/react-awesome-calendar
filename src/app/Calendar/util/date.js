export const getDate = date => {
  if (checkDate(date)) {
    return new Date(date.getFullYear(), date.getMonth(), date.getDate());
  }
};

export const getTimeFromDate = date => {
  const hoursString = date.getHours().toString();
  const minsString = date.getMinutes().toString();
  let hour = hoursString;
  if (hoursString.length < 2) {
    hour = '0' + hoursString;
  }
  let mins = minsString;
  if (minsString.length < 2) {
    mins = '0' + minsString;
  }

  return hour + ':' + mins;
};

export const checkDate = date =>
  Object.prototype.toString.call(date) === '[object Date]';
