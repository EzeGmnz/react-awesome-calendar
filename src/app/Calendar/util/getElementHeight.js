export const getElementHeight = (element) => {
    let height = element.offsetHeight;
    height += parseInt(window.getComputedStyle(element).getPropertyValue('margin-top'));
    height += parseInt(window.getComputedStyle(element).getPropertyValue('margin-bottom'));

    return height;
};