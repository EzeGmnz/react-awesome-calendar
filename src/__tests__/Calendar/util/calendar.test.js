import { getCalendarMonth, formatEvents, getEventsForCalendar, getMonthName } from '../../../app/Calendar/util/calendar';

describe('calendar', () => {
  describe('getCalendarMonth', () => {
    it('should return a month from 28th April - 1st June', () => {
      const month = 4;
      const year = 2019;
      expect(getCalendarMonth(month, year)).toMatchSnapshot();
    });
  });
});