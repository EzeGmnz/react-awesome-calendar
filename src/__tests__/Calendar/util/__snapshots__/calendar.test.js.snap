// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calendar getCalendarMonth should return a month from 28th April - 1st June 1`] = `
Array [
  Array [
    Object {
      "current": false,
      "date": 2019-04-27T23:00:00.000Z,
    },
    Object {
      "current": false,
      "date": 2019-04-28T23:00:00.000Z,
    },
    Object {
      "current": false,
      "date": 2019-04-29T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-04-30T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-01T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-02T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-03T23:00:00.000Z,
    },
  ],
  Array [
    Object {
      "current": true,
      "date": 2019-05-04T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-05T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-06T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-07T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-08T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-09T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-10T23:00:00.000Z,
    },
  ],
  Array [
    Object {
      "current": true,
      "date": 2019-05-11T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-12T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-13T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-14T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-15T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-16T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-17T23:00:00.000Z,
    },
  ],
  Array [
    Object {
      "current": true,
      "date": 2019-05-18T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-19T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-20T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-21T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-22T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-23T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-24T23:00:00.000Z,
    },
  ],
  Array [
    Object {
      "current": true,
      "date": 2019-05-25T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-26T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-27T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-28T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-29T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-30T23:00:00.000Z,
    },
    Object {
      "current": false,
      "date": 2019-05-31T23:00:00.000Z,
    },
  ],
]
`;
